lib/go/edgecontext/backend.go

```diff
@@ -0,0 +1,124 @@
+package edgecontext
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/reddit/baseplate.go/ecinterface"
+	v0 "github.com/reddit/edgecontext/lib/go/internal/v0"
+)
+
+type Backend struct {
+	SetEdgeContext func(context.Context, *EdgeRequestContext) context.Context
+	GetEdgeContext func(context.Context) (*EdgeRequestContext, bool)
+	Factory        func(Config) ecinterface.Factory
+	Init           func(Config) *Impl
+	New            func(context.Context, *Impl, NewArgs) (*EdgeRequestContext, error)
+	FromHeader     func(context.Context, string, *Impl) (*EdgeRequestContext, error)
+}
+
+var defaultBackend = &Backend{
+	SetEdgeContext: func(ctx context.Context, ec *EdgeRequestContext) context.Context {
+		if ec == nil {
+			return ctx
+		}
+		return v0.SetEdgeContext(ctx, ec.Source.(*v0.DataSource))
+	},
+	GetEdgeContext: func(ctx context.Context) (*EdgeRequestContext, bool) {
+		source, ok := v0.GetEdgeContext(ctx)
+		if !ok {
+			return nil, false
+		}
+		return &EdgeRequestContext{
+			Source: source,
+			ctx:    ctx,
+		}, true
+	},
+	Factory: func(cfg Config) ecinterface.Factory {
+		return v0.Factory(v0.Config{
+			Store:  cfg.Store,
+			Logger: cfg.Logger,
```

Comment:
:no_bell: we should move away from this and use slog to log directly instead. (can do that in follow up PRs instead)

Reply:
I'll go ahead and do that now, lets me take it out of the interface.